src/components/firebase/AnalyticsProvider.jsx

```diff
@@ -0,0 +1,23 @@
+import { getAnalytics } from 'firebase/analytics';
+import PropTypes from 'prop-types';
+import { createContext, useContext } from 'react';
+import { useFirebaseApp } from './FirebaseAppProvider';
+
+const AnalyticsSdkContext = createContext(undefined);
+
+export function AnalyticsProvider(props) {
+  const app = useFirebaseApp();
+
+  let { sdk } = props;
+
+  if (!sdk) {
+    sdk = getAnalytics(app);
+  }
+
+  return <AnalyticsSdkContext.Provider value={sdk} {...props} />;
+}
+AnalyticsProvider.propTypes = {
+  sdk: PropTypes.object,
+};
+
+export const useAnalytics = () => useContext(AnalyticsSdkContext);
```

Comment:
A named function might make stack traces easier to debug here.
```suggestion
export function useAnalytics() {
	return useContext(AnalyticsSdkContext);
}
```

Reply:
I'm starting to think if we aren't sending custom events that we don't really need a provider...